### ISSUE TITLE
Cumulative Profit % 

### ISSUE THREAD

**Issue by tjoshi01** [freqtrade/freqtrade#9109]
Hello,

Is there any way to get a monthly breakdown of profits - stating the exact amount staked during each month? 

As I understand,  "Tot Profit %" is calculated against the initial balance. How is it that the final balance does not reflect the unstaked portion of the initial balance? I want to see the average stake amount for each month. Is there any way to do so? 

Thanks in advance!

**Comment by xmatthias** [comment]
we have rpc commands for this - namelly `/daily`, `/weekly` and `/monthly`.

the [documentation](https://www.freqtrade.io/en/stable/telegram-usage/#daily) will have some examples for this.

**Comment by tjoshi01** [comment]
@xmatthias, I found below notes on interpreting backtesting results:

- "Cum Profit % sums up all the profits/losses."
-  "Tot Profit % shows the total profit % in relation to the starting balance."

I am unable to generate Cum Profit % in the month-by-month report. The program outputs wins, losses, draws and Tot Profit %, but not CP %. 

If I am not mistaken, Cum Profit % can be seen as the upper bound on profits. Total Profit % can be a good metric iff the average stake amount is close to $1000.

**Comment by xmatthias** [comment]
it's actually unclear what we're talking about here.

for a running bot - the rpc methods mentioned above will be what you need, and will contain the profit the bot made that timeperiod (obviously calculated against the starting balance for that timeperiod).

for backtesting, the same applies for the output depending on the selected breakdown.

**Comment by tjoshi01** [comment]
@xmatthias, I see. Thanks a lot. I have used dynamic staking & there is a big gap between cumulative profit and final profit. It turns out that compounding does not always mean more money!

**Comment by xmatthias** [comment]
well it's still unclear what you're talking about (trade, backtest, ...) - so i guess you'll have found an answer yourself.

**Comment by tjoshi01** [comment]
@xmatthias, let me use an example. I really appreciate any help you can give me - a strategist!

1. Suppose I make 4 trades (starting at $1000) and they give me: _10%, 20%, 10%, -25%_, then my cumulative profit is the sum of these four values: 15%. 

2. However, since -25% is associated with a large stake, my total profit comes out to be only 9%. This is much less than 15%.

I don't know how to configure the bot such that, what's known as cumulative profit is exactly equal to total profit. You may think of this and wonder what my backtesting configuration could be:

In the configuration, if the "stake_amount" parameter is set to "unlimited", then there will be large differences between cumulative and total profits, and asset managers will be dissuaded from investing in my strategy :(

However if "stake_amount" is set to $1000, the bot simply stops trading if I end up with less than $1000 in my account. I guess I need something like this: 
`if (balance >= 1000) : stake_amount = 1000; else : stake_amount = balance; `

**Comment by xmatthias** [comment]
> 10%, 20%, 10%, -25%, then my cumulative profit is the sum of these four values: 15%.

this is wrong, unless all stakes are identical. 
You can't aggregate relative values like this - because 
10% is relative to 1000
20% is relative to 1100
10% is relative to 1320
-25% is relative to 1452

There's no "right or wrong" configuration - you can either use compounding (stake-amount = unlimited) - and get increasing (or decreasing) stakes - or you can skip that (stake_amount = fixed), effectively leaving gains on the table (so all values are relative to the initial 1000$).

If you carefully read the documentation, you'll notice that there are ways to "amend the last stake amount".

---

On a side note: if 1 loss can offset 2 wins easily, then your R:R is off, and no asset manager would invest in your strategy anyway.

**Comment by tjoshi01** [comment]
I think you are right about identical stakes, in that, to reduce the deviation between sum of profits and total profits, an identical stake amount is the way to go.

Btw, I found the part about stake amendments. Cheers! This is exactly what I needed to ensure that profits are not undercut by unlimited stakes (and the tendency of larger prices to fall by greater margins). Nevermind! Thanks a lot for weighing in :)
